src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -865,15 +865,23 @@ public BlittableJsonReaderObject SetTopology(TransactionOperationContext context
             return topologyJson;
         }
         
-        public void NotifyTopologyChange()
+        public void NotifyTopologyChange(bool propogateError = false)
         {
-            if(IsDisposed)
-                return;
-            
-            using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
-            using (ctx.OpenReadTransaction())
+            try
+            {
+                if (IsDisposed)
+                    return;
+                
+                using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                using (ctx.OpenReadTransaction())
+                {
+                    TopologyChanged?.Invoke(this, GetTopology(ctx));
+                }
+            }
+            catch (Exception)
             {
-                TopologyChanged?.Invoke(this, GetTopology(ctx));
+                if (propogateError)
```

Comment:
Probably a good idea to have a log here, so we won't swallow the exception entirely.